text/0000-anonymous-variants.md

```diff
@@ -0,0 +1,306 @@
+
+Parsing such a type is similarly easy. An anonymous variant type names consists of a parentheses-enclosed vertical bar separated list of types, optionally with a trailing vertical bar. At the level of a token stream, which will group any nested anonymous variant types into a single token tree, the stream can be divided into potential type names at the vertical bars, but not including the bars themselves, the last one removed if it consists of exactly zero tokens (to account for trailing vertical bars), and the remaining token stream excerpts either subsequently parsed to see if each one corresponds to a type, or just kept as is if semantic analysis is not desired. 
+
+Here is an example bringing all the above together, showing how one might automatically derive a trait for an anonymous variant type. Note that this is only a demonstration summarizing the above points, and it omits a number of checks and extras that a production-grade procedural macro would have, but it should work if a token group corresponding to a valid name of an anonymous variant type whose variants all implement Debug is passed into `debug_derive`. 
+```
+fn debug_derive(typename: Group) -> TokenStream {
+    // Copy the full type name to a string
+    let fulltype = typename.clone();
+    // Count the number of token stream parts corresponding to a type
+    let mut token_count = 0_u64;
+    let mut type_count = 0_u64;
+    for piece in typename.stream().into_iter() {
+        match piece {
+            TokenTree::Punct(p) if p.as_char() == '|' => {
+                if token_count > 0 {
+                    token_count = 0;
+                    type_count += 1;
+                } else {
+                    panic!("Not a valid anonymous variant type");
+                }
+            },
+            _ => {token_count += 1; }
+        }
+    }
+    if token_count > 0 {
+        type_count += 1;
+    }
+    // Map the valid indices to match arms
+    let match_arms = (0..type_count)
+        .flat_map(|index| (format!{"_:: {} (v) => f.write_fmt(
+            format_args!(\"{{}}({{:?}})\", {}, v)
+        ),", index, index}).parse::<TokenStream>().unwrap())
+        .collect::<TokenStream>();
+    // Put it all together
+    (format!{"impl ::std::fmt::Debug for {} {{
+        fn fmt(&self, f: &mut ::std::fmt::Formatter) -> 
+            Result<(), ::std::fmt::Error> 
+        {{
+            match self {{ {} }}
+        }}
+    }}", fulltype, match_arms}).parse::<TokenStream>().unwrap()
+}
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A semi-formal description of the syntax of anonymous variant types is shown below, where "..." represents the current rust grammar, immediately before implementation of this RFC. (Thanks to [Centril](https://github.com/Centril) for pointing out a simplification of the grammar, among other things) 
```

Comment:
Thanks! But there's no need to mention me in the text and it takes focus away from the proposal. :)

Reply:
Removed with the next commit.